guides/source/caching_with_rails.md

```diff
@@ -387,10 +387,10 @@ gem 'connection_pool'
 Next, pass the `:pool_size` and/or `:pool_timeout` options when configuring the cache store:
 
 ```ruby
-config.cache_store = :mem_cache_store, "cache.example.com", { pool_size: 5, pool_timeout: 5 }
+config.cache_store = :mem_cache_store, "cache.example.com", { pool_size: ENV.fetch("RAILS_MAX_THREADS") { 5 }, pool_timeout: 5 }
 ```
 
-* `:pool_size` - This option sets the number of connections per process (defaults to 5).
+* `:pool_size` - This option sets the number of connections per process and it is recommended to set the value to the number threads configured on the threaded server.
```

Comment:
This is what I had in the original PR but it was rejected. See discussion here:
https://github.com/rails/rails/pull/37393#discussion_r361872708

Reply:
I don’t seem to be able to see the original comment. Should we not recommend what the default should be? The default value here is an internal detail of connection_pool so I figured setting the pool size should be more intentional by the user.

Reply:
<img width="686" alt="image" src="https://user-images.githubusercontent.com/28561/85012644-9cdf9b00-b163-11ea-9e33-10bd266eb2e8.png">

Reply:
It isn't being said it words but Rails indirectly recommends it 

https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/config/databases/postgresql.yml.tt#L22

Maybe something like  

> This option sets the number of connections per process and a good starting default would be to set a pool size that is equal to the number of threads running on each process. 

cc @eugeneius

Reply:
Active Record connections are checked out until the end of the current request, so in practice the pool size typically does need to match the number of application threads to avoid contention. Dalli connections are only checked out for each operation, so unless your application spends a significant chunk of its time reading from the cache, it's unlikely to need a pool size that high. That's why "It is recommended" got my hackles up in https://github.com/rails/rails/pull/37393: I suspect very few applications get tangible benefit from connection pooling at all, and even for those that do, five connections is probably enough.

That said, I'm fine with suggesting that configuration here as a reasonable default, as long as we avoid the phrase "it is recommended". 🙂 I'd keep the note about the default value though, since it's possible to enable connection pooling by specifying the timeout without a pool size.